Automattic/newspack-plugin
# Feat / IA for Newsletters

### All Submissions:






### Changes proposed in this Pull Request:



Once the "TODOs" below are done, this PR will complete the Information Architecture Wizard for the Newsletters Plugin. This code will update the existing Newsletters Plugin header and menu.  

See screenshot:

![newsletters-tabs](https://github.com/user-attachments/assets/e1410164-e26c-46b8-b73a-91d90c85583f)

* The Newsletters custom blue header is deactivated.
* A new Admin Header is added.
* The menu is moved and compressed.
* Some menu items are now combined into Tabs.

Additionally the new Dashboards page is updated so that if the Newsletters plugin is not installed, then the dashboard will not show Newsletters items.  

See screenshot:

![newsletters-dashboard-new](https://github.com/user-attachments/assets/30cc5859-84ad-4341-8fee-c1bd7246a774)


### TODOs:

- We may need to change the menu icon.  Export the SVG from Figma or talk to Thomas/designer.  See code comment: `// @TODO get SVG from Figma? This one is "envelope" from: https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library`

- I was having a hard time setting the Newsletters CPT parent menu item to a decimal value.  It seems that CPTs only support an integer value for `menu_positon`.  So I created a function that will loop through `global $menu` and then move the CPT to position 3.3 or "3.33" if something is already at 3.3.  See the current code in function `move_cpt_menu()`.  You may want to look at function `registered_post_type_newsletters( $post_type )` to see how I'm changing the `menu_icon` but when I try to change the `menu_position` to 3.3 in that function it doesn't work.

- I didn't have chance to fully test what happens if someone goes to a "bookmarked" category or tag page that was removed by this Wizard.  I recommended we remove categories and tags from the Newsletters menu in this PR ( https://github.com/Automattic/newspack-newsletters/pull/1678 ) but still, even with the links removed, someone could have bookmarked any of the following links, so we may want to view them in the browser and see what the user would see.  We'll probably want to remove the "blue" Newsletters header and apply the new Admin Header like this code does for the other screens.  
  - edit-tags.php?taxonomy=category&post_type=newspack_nl_cpt
  - edit-tags.php?taxonomy=post_tag&post_type=newspack_nl_cpt
  - term.php?taxonomy=category&tag_ID=4&post_type=newspack_nl_cpt
  - term.php?taxonomy=post_tag&tag_ID=31&post_type=newspack_nl_cpt

- There is an existing Newsletters admin screen in the current (trunk) Newspack Plugin. It can be seen in wp-admin `Newspack > Engagement > Newsletters` or `admin.php?page=newspack-engagement-wizard#/newsletters`.  It looks to be a React version of the existing Newsletters Settings screen.  Typically I'm not a fan of having different screens that do the same thing, so ... maybe we can just remove the React version when IA goes live since Newsletters already has a Settings screen?  But I'm not in charge 😉 If it is kept, I'm not sure how to "plug it in" so that admin users that have the Newspack Plugin with IA active get "redirected/shown" the React version of the screen vs admin users that just use Newsletters stand-alone without the Newspack Plugin who will just see the normal settings page.

- Finally, I'm not sure why `ci/cirlceci:test-php` is failing in github below.  Maybe it was an error that got merged into this branch via `trunk` or `epic/ia`?

### How to test the changes in this Pull Request:

1. Pull branch.
2. `npm run build`
3. Activate Newspack plugin.
4. Install/activate Newspack Newsletters Plugin.
5. Verify screenshots and notes above.
6. Deactivate Newsletters plugin, verify dashboard no longer shows Newsletters section.
7. Re-activate the Newsleters plugin and do the one test related to `get_settings_url()` in the related PR ( https://github.com/Automattic/newspack-newsletters/pull/1678 )

### Other information:
